### PR TITLE
add telemetry to muddle register

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: libs/muddle/src/muddle_register.cpp [hunk 1]
```diff
@@ -301,6 +339,7 @@ Address MuddleRegister::GetAddress(ConnectionHandle handle) const
  */
 void MuddleRegister::Enter(WeakConnectionPtr const &ptr)
 {
+  connections_entered_total_->add(1);
```
Comment (ejfitzgerald) [inline libs/muddle/src/muddle_register.cpp]:
```suggestion
  connections_entered_total_->increment();
```

File: libs/muddle/src/muddle_register.cpp [hunk 2]
```diff
@@ -333,6 +372,8 @@ void MuddleRegister::Enter(WeakConnectionPtr const &ptr)
  */
 void MuddleRegister::Leave(ConnectionHandle handle)
 {
+  connections_left_total_->add(1);
```
Comment (ejfitzgerald) [inline libs/muddle/src/muddle_register.cpp]:
```suggestion
  connections_left_total_->increment();
```

File: libs/muddle/src/muddle_register.cpp [hunk 3]
```diff
@@ -370,6 +411,7 @@ void MuddleRegister::Leave(ConnectionHandle handle)
   // signal the router
   if (callback_copy)
   {
+    connections_callbacks_called_total_->add(1);
```
Comment (ejfitzgerald) [inline libs/muddle/src/muddle_register.cpp]:
```suggestion
    connections_callbacks_called_total_->increment();
```

File: libs/muddle/internal/muddle_register.hpp [hunk 4]
```diff
@@ -115,13 +115,21 @@ class MuddleRegister : public network::AbstractConnectionRegister
   /// @}
 
 private:
+  using GaugePtr   = telemetry::GaugePtr<uint64_t>;
```
Comment (ejfitzgerald) [inline libs/muddle/internal/muddle_register.hpp]: `GaugePtr` doesn't look like it is being used

**Review by ejfitzgerald — COMMENTED** [review]
Very minor things to be honest - I believe increment will be marginally more performance because of the pre-increment on the atomic

File: libs/muddle/internal/muddle_register.hpp (hunk 4 above)
Comment (n-hutton) [reply libs/muddle/internal/muddle_register.hpp]: ok

File: libs/muddle/src/muddle_register.cpp (hunk 3 above)
Comment (n-hutton) [reply libs/muddle/src/muddle_register.cpp]: ok

File: libs/muddle/src/muddle_register.cpp (hunk 2 above)
Comment (n-hutton) [reply libs/muddle/src/muddle_register.cpp]: ok

File: libs/muddle/src/muddle_register.cpp (hunk 1 above)
Comment (n-hutton) [reply libs/muddle/src/muddle_register.cpp]: ok

File: libs/telemetry/src/registry.cpp [hunk 5]
```diff
@@ -90,6 +90,10 @@ CounterPtr Registry::CreateCounter(std::string name, std::string description, La
       measurements_.push_back(counter);
     }
   }
+  else
+  {
+    std::cout << LOGGING_NAME << ": Failed to validate telemetry name: " << name << std::endl;
```
Comment (ejfitzgerald) [inline libs/telemetry/src/registry.cpp:95]: Not 100% keen on having multiple things talking to `cout` at the same time. Was this just debug?

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**